### PR TITLE
Microverse Ruby Project 1 - Bubble Sorts Method

### PR DESCRIPTION
**Pull request by mahmoud717**
For this project, we did the following:
1- created two **sorting algorithms** that use the [bubble sort](https://en.wikipedia.org/wiki/Bubble_sort) method.
2- the bub_sort algorithm can sort any given array of numbers In ascending order.
3- the bub_sort_by algorithm can sort any given array of strings by the string length in ascending order.

### CODE REVIEW

File: bubble_sort.rb [hunk 1]
```diff
@@ -1,33 +1,31 @@
...
-       break if  swapped == false 
+  loop do
+    swapped = false
+    (0..array.length - 2).each do |x|
+      if array[x] > array[x + 1]
+        array[x], array[x + 1] = array[x + 1], array[x]
+        swapped = true
+      end
     end
+    print array
```
Comment (ikraamg) [inline bubble_sort.rb]: Please remove the print statement and return your array at the end of the method.

File: bubble_sort.rb [hunk 2]
```diff
@@ -1,33 +1,31 @@
...
-       break if  swapped == false 
+  loop do
+    swapped = false
+    (0..array.length - 2).each do |x|
+      if yield(array[x], array[x + 1]).positive?
+        array[x], array[x + 1] = array[x + 1], array[x]
+        swapped = true
+      end
     end
+    print array
```
Comment (ikraamg) [inline bubble_sort.rb]: Please remove the print statement and return your array at the end of the method

**Review by ikraamg — CHANGES_REQUESTED** [review]
Hi @mahmoud717 and @gurbuzertunga 

### **Status: Almost there! Minor changes required** :recycle: 

Great work! Your bubble sort algorithm has the correct logic and you have a good readme file.

A couple of things need to be fixed before we can have this merged: 

- [ ] Please ensure you write descriptive commit messages, this is very important for code history both as an individual and within a team.

- [ ] Please be a bit more descriptive in your PR descriptions as this gives context to the code reviewers. Very important when working within a team. 

- [ ] This might feel pedantic but the methods don't actually return anything. They print the array to the console but if you called these functions as part of a larger program, the functions would just return nil rather than the sorted array.
- [ ] Please remove the 'print' and create a return statement at the end of your function instead. To test if this works, you can call the method `puts bub_sort(arr)` and it should print to the console without having a print statement inside your method. 
- As a general note, remember to not have any 'debugging' code in your final PR such as printing to the console. 

If you have any questions, don't hesitate to contact me through slack: **_@ikraam_**

_**I wish you all the best!**_  :desktop_computer: 

_Project reviewed by Ikraam Ghoor (TSE)_

File: bubble_sort.rb [hunk 3]
```diff
@@ -1,33 +1,33 @@
...
+        swapped = true
+      end
     end
+    break if swapped == false
+  end
+  array
 end
-bub_sort_by(["hey", "hi", "hello"]){|left, right| left.length - right.length}
\ No newline at end of file
+p bub_sort_by(%w[hey hi hello]) { |left, right| left.length - right.length }
```
Comment (elmejdki) [inline bubble_sort.rb]: please remove all the debugging code and leave only your functions

File: bubble_sort.rb [hunk 4]
```diff
@@ -1,33 +1,33 @@
...
+    break if swapped == false
+  end
+  array
 end
-bub_sort([11 ,2 ,1 ,23 ,10 , 8, 3])
+
+p bub_sort([11, 2, 1, 23, 10, 8, 3])
 ##############################
 ##############################
 ##############################
```
Comment (elmejdki) [inline bubble_sort.rb]: remove this also

**Review by elmejdki — CHANGES_REQUESTED** [review]
# status: require changes :recycle: 
## Great job so far @team, you have just some minor changes to work :toolbox: :wrench: :hammer: on to get your project approved :checkered_flag:

- [ ] please remove all debugging code and leave only the functions.

### Please make all the required changes before use submit another code review.
## Happy Coding :tophat:

**Comment by mahmoud717** [comment]
didn't know that this was debugging code ( ;

**Review by elmejdki — APPROVED** [review]
# status: approved :heavy_check_mark: 
## Hi, it's me again,

I think you didn't understand me, you should delete the print methods, we should have only the functions inside the file, I have approved your project but please make that happen.

## Happy Coding :tophat:

**Pull request merged**